.github/workflows/semantic.yml

```diff
@@ -0,0 +1,33 @@
+# Check that the commit messages are between 20 and 150 chars,
+# and that they follow the rule
+# <type> (<scope>): <subject>
+# type: docs, feat, fix, refactor, style or test
+# scope (optional): any extra info, (like DMS or whatever)
+
+name: 'Commit Message Check'
+on: pull_request
+
+jobs:
+  check-commit-message:
+    name: Check Commit Message
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Commit Format
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^(docs|feat|fix|refactor|style|test)( ?\(.*\))?: .+$'
+          excludeDescription: 'true' # optional: this excludes the description body of a pull request
+          excludeTitle: 'true' # optional: this excludes the title of a pull request
+          checkAllCommitMessages: 'true' # optional: this checks all commits associated with a pull request
+          accessToken: ${{ secrets.GITHUB_TOKEN }} # github access token is only required if checkAllCommitMessages is true
+          flags: 'gm'
+          error: 'Your commit has to follow the format "<type>(<scope>): <subject>"".'
+      - name: Check Commit Length
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^.{20,150}$'
```

Comment:
This only limits the length of the first line of the commit, or also the total length of the commit message?

I  occasional longer commit message, and many times multiple lines

Reply:
I just tested it (https://github.com/chaen/DIRAC/pull/20/checks?check_run_id=3610068989), and it limits the first line, as of now. To be honest, long commits are very rare. And it's more the `fix`, `typo`, `test` commits that I want to avoid....

And looking at your commit, the longest I could find was 155 char 

```
commit fc08962f39d9abfcd61c0bfb90cc8e61c6076127
Author: Andre Sailer <andre.philippe.sailer@cern.ch>

    FileCatalogDirectoryTreeBase: prevent RuntimeError (maximum recursion depth exceeded) in FileCatalog service when illegal path is given to some functions.
```

Reply:
If it only limits the first line, this is OK.

I was thinking of commits like this 99da524399ff62f6611df97c9775ce1973cda2b8